docs/admin_api/user_admin_api.md

```diff
@@ -1155,3 +1155,27 @@ GET /_synapse/admin/v1/username_available?username=$localpart
 
 The request and response format is the same as the
 [/_matrix/client/r0/register/available](https://matrix.org/docs/spec/client_server/r0.6.0#get-matrix-client-r0-register-available) API.
+
+### Find a user based on their ID in an auth provider
+
+The API is:
+
+```
+GET /_synapse/admin/v1/auth_providers/$provider/users/$external_id
+```
+
+A response body like the following is returned:
+
+```json
+{
+    "user_id": "@hello:example.org"
+}
+```
+
+**Parameters**
+
+The following parameters should be set in the URL:
+
+- `provider` - The ID of the authentication provider, as advertised by the [`GET /_matrix/client/v3/login`](https://spec.matrix.org/latest/client-server-api/#post_matrixclientv3login) API in the `m.login.sso` authentication method.
+- `external_id` - The user ID from the authentication provider. Usually corresponds to the `sub` claim for OIDC providers, or to the `uid` attestation for SAML2 providers.
```

Comment:
Are typical values for these params URL-safe? If not, might we need to warn admins to urlencode them?

Reply:
They are, since they are the same as the ones advertised in the `m.login.sso` login methods, under `> identity_providers > id`, which is, [according to the spec](https://spec.matrix.org/v1.3/client-server-api/#client-login-via-sso):

> Opaque string chosen by the homeserver, uniquely identifying the IdP from other IdPs the homeserver might support. Should be between 1 and 255 characters in length, containing unreserved characters under [RFC 3986](http://www.ietf.org/rfc/rfc3986.txt) (ALPHA DIGIT "-" / "." / "_" / "~"). Clients are not intended to parse or infer meaning from opaque strings.

But I'm not sure there is anything really enforcing that in Synapse?

Reply:
`sub` claims could indeed not be url-safe, and can typically include `/`, `@` and `:`. I'll mention that in the doc